R/LearnerPythonClassif.R

```diff
@@ -0,0 +1,166 @@
+# R/learner_python_classif_base.R
+
+LearnerPythonClassif = R6::R6Class(
+  "LearnerPythonClassif",
+  inherit = mlr3::LearnerClassif,
+
+  public = list(
+    initialize = function(id,
+                          feature_types   = c("logical","integer","numeric","factor","ordered"),
+                          predict_types   = c("response", "prob"),
+                          param_set       = paradox::ParamSet$new(),
+                          properties      = character(),
+                          packages        = "reticulate",              # R packages
+                          label           = NA_character_,
+                          man             = NA_character_,
+                          # Python requirements
+                          py_packages,
+                          python_version,
+                          method          = c("auto","virtualenv","conda"),
+                          envname         = NULL) {
+
+      method = match.arg(method)
+
+      base_ps = ps(
+        py_env    = p_uty(default = envname, tags = "python"),
+        py_method = p_fct(levels = c("auto","virtualenv","conda"),
+                          default = method, tags = "python")
+      )
+      param_set = ps_union(list(param_set, base_ps))
+
+      super$initialize(
+        id = id,
+        feature_types = feature_types,
+        predict_types = predict_types,
+        param_set     = param_set,
+        properties    = properties,
+        packages      = union(c("mlr3", "mlr3extralearners"), packages),
+        label = label,
+        man   = man
+      )
+
+      private$.py_packages = py_packages
+      private$.py_version  = python_version
+      private$.py_method   = method
+      private$.py_env      = envname
+    },
+
+    py_requirements = function() {
+      list(
+        packages       = private$.py_packages,
+        python_version = private$.py_version,
+        method         = private$.py_method,
+        envname        = private$.py_env
+      )
+    },
+
+    ensure_deps = function() {
+      if (!requireNamespace("reticulate", quietly = TRUE)) {
+        stop("Package 'reticulate' required.")
+      }
+      # assert/install in-session
+      reticulate::py_require(private$.py_packages, python_version = private$.py_version)
+      invisible(TRUE)
+    }
+  ),
+
+  private = list(
+    .py_packages = NULL, .py_version = NULL, .py_method = NULL, .py_env = NULL,
+
+    .train = function(task) {
+      self$ensure_deps()
+
+      x    = task$data(cols = task$feature_names)
+      y    = task$truth()
+      pars = self$param_set$get_values(tags = "train")
+
+      fit = private$.fit_py(x = x, y = y, pars = pars, task = task)
```

Comment:
Keep `$.fit_py()` but only pass `task`. Sometimes you want to get `pars` with a different `tag` and `X` and `y` also does not work for every learner.